src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -808,16 +782,6 @@ public BaseResponseBean advanceSessionWithEndpoint(MenuSession menuSession, Stri
         // Sync requests aren't run when executing operations, so stop and check for them after each operation
         for (StackOperation op : endpoint.getStackOperations()) {
             sessionWrapper.executeStackOperations(new Vector<>(Arrays.asList(op)), evalContext);
-            FormplayerSyncScreen screen = menuSession.getNextScreenIfSyncScreen(false, new EntityScreenContext());
-            if (screen != null) {
-                try {
-                    screen.init(sessionWrapper);
-                    doPostAndSync(menuSession, (FormplayerSyncScreen)screen);
-                    executeAndRebuildSession(menuSession);
```

Comment:
I don't think `executeAndRebuildSession` is needed anymore since the sync screen is now handled while the session is being rebuilt. But i'm not completely confident, @shubham1g5 do you know?

Reply:
Agree that it should not be needed anymore.